### PR TITLE
[CatalogPromotion] Introduce CatalogPromotionCreated event

### PR DESCRIPTION
**Pull request by GSadee**
| Q               | A
| --------------- | -----
| Branch?         | 1.11
| Bug fix?        | no
| New feature?    | no?
| BC breaks?      | no
| Deprecations?   | 
| Related tickets | 
| License         | MIT

### CODE REVIEW

**Review by AdamKasp — APPROVED** [review]

File: docs/book/products/catalog_promotions.rst [hunk 1]
```diff
@@ -317,8 +317,9 @@ How the Catalog Promotions are applied?
 ---------------------------------------
 
 The Catalog Promotion application process utilises `API Platform events <https://api-platform.com/docs/core/events/>`_ for an API.
-and `Resource events </book/architecture/events>`_ for UI. When a new Promotion is created or the existing one is edited
-there are services that listen on proper events and dispatch ``CatalogPromotionUpdated`` event to event bus.
+and `Resource events </book/architecture/events>`_ for UI. When a new Promotion is created there are services that listen
```
Comment (SirDomin) [inline docs/book/products/catalog_promotions.rst:320]:
```suggestion
And `Resource events </book/architecture/events>`_ for UI. When a new Promotion is created there are services that listen
```

**Review by SirDomin — APPROVED** [review]

File: docs/book/products/catalog_promotions.rst (hunk 1 above)
Comment (GSadee) [reply docs/book/products/catalog_promotions.rst:320]: Fixed 👌🏻

File: src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php [hunk 2]
```diff
@@ -40,21 +41,45 @@ public function __construct(
...
+        /** @var CatalogPromotionInterface $catalogPromotion */
+        $catalogPromotion = $event->getSubject();
+        Assert::isInstanceOf($catalogPromotion, CatalogPromotionInterface::class);
+
+        $this->eventBus->dispatch(
+            new CatalogPromotionCreated($catalogPromotion->getCode()),
+            $this->calculateStartDateStamp($catalogPromotion)
+        );
+
+        $this->dispatchCatalogPromotionEndedEvent($catalogPromotion);
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/EventListener/CatalogPromotionEventListener.php:55]: I'm wondering if in the next iteration we couldn't extract this logic and reuse it here and in Api Platform context. Same for update action

**Review by lchrusciel — APPROVED** [review]

**Comment by lchrusciel** [comment]
Thanks, Grzegorz! :1st_place_medal:

**Pull request merged**